### ISSUE TITLE
`en_ner_bionlp13cg_md` Label names are wrong?

### ISSUE THREAD

**Issue by FrancescoCasalegno** [allenai/scispacy#251]
#### First possible list of labels
As far as I understand, the scispacy model `en_ner_bionlp13cg_md` should be trained on a corpus with 16 possible annotation labels (see e.g. [here](https://arxiv.org/pdf/1911.03869v3.pdf) for this list):
1. AMINO ACID
2. ANATOMICAL SYSTEM
3. CANCER
4. CELL
5. CELLULAR COMPONENT
6. CHEMICAL
7. DEVELOPING ANATOMICAL STRUCTURE
8. GENE/PROTEIN
9. IMMATERIAL ANATOMICAL
10. MULTI-TISSUE STRUCTURE
11. ORGAN
12. ORGANISM
13. ORGANISM SUBDIVISION
14. ORGANISM SUBSTANCE
15. PATHOLOGICAL FORMATION
16. TISSUE

Am I right in saying that `en_ner_bionlp13cg_md` predicts these 16 labels (just with different names)?

#### Second possible list of labels
However, if we run `spacy.load("en_ner_bionlp13cg_md").entity.labels` we get the following 17 labels
1. AMINO_ACID
2. ANATOMICAL_SYSTEM
3. CANCER
4. CELL
5. CELLULAR_COMPONENT
6. DEVELOPING_ANATOMICAL_STRUCTURE
7. ENTITY
8. GENE_OR_GENE_PRODUCT
9. IMMATERIAL_ANATOMICAL_ENTITY
10. MULTI
11. ORGAN
12. ORGANISM
13. ORGANISM_SUBDIVISION
14. ORGANISM_SUBSTANCE
15. PATHOLOGICAL_FORMATION
16. SIMPLE_CHEMICAL
17. TISSUE

This second list looks wrong to me. Indeed, after running the model on a few texts, I found out that the model actually never predicted `MULTI` or `TISSUE` labels (which however appear in the list), but did instead predict `MULTI-TISSUE_STRUCTURE` (which does not appear in the list!).

How is this possible? If `ner.entity.labels` is wrong, how can the predicted annotations be right?
Am I missing something?

#### Third possible list of labels
Finally, the list of entity types mentioned in the [scispacy docs|https://allenai.github.io/scispacy/] also look wrong to me. It mentions 14 labels:
1. AMINO_ACID
2. ANATOMICAL_SYSTEM
3. CANCER
4. CELL
5. CELLULAR_COMPONENT
6. DEVELOPING_ANATOMICAL_STRUCTURE
7. GENE_OR_GENE_PRODUCT
8. IMMATERIAL_ANATOMICAL_ENTITY
9. MULTI-TISSUE_STRUCTURE
10. ORGAN
11. ORGANISM
12. ORGANISM_SUBDIVISION
13. SIMPLE_CHEMICAL
14. TISSUE

For instance, `PATHOLOGICAL_FORMATION` is missing from this list.

**Comment by danielkingai2** [comment]
Thanks for the report!
The list of labels present in the dataset is:

```{'AMINO_ACID',
 'ANATOMICAL_SYSTEM',
 'CANCER',
 'CELL',
 'CELLULAR_COMPONENT',
 'DEVELOPING_ANATOMICAL_STRUCTURE',
 'GENE_OR_GENE_PRODUCT',
 'IMMATERIAL_ANATOMICAL_ENTITY',
 'MULTI-TISSUE_STRUCTURE',
 'ORGAN',
 'ORGANISM',
 'ORGANISM_SUBDIVISION',
 'ORGANISM_SUBSTANCE',
 'PATHOLOGICAL_FORMATION',
 'SIMPLE_CHEMICAL',
 'TISSUE'}```

This is basically your lists #1 (with some small changes to the exact names). List #2 contains "ENTITY" because we loaded the generic NER pipe first when training this model, this should not affect the output. I am not sure why spacy splits up `MULTI-TISSUE_STRUCTURE` when displaying `nlp.entity.labels`, perhaps this is just some formatting thing, like maybe spacy expects no hyphens in the labels or something. I'd have to look into that more, but since you said the model does predict `MULTI-TISSUE_STRUCTURE`, I'm guessing this is not affecting the model either. The third list is just wrong (it is missing `PATHOLOGICAL_FORMATION` and `ORGANISM_SUBSTANCE`), and we need to fix that, thanks!

**Comment by FrancescoCasalegno** [comment]
Dear @danielkingai2 thanks for the prompt reply.

> perhaps this is just some formatting thing, like maybe spacy expects no hyphens in the labels or something

Yes, I just found the reason why your `model.entity.labels` is broken. The reason is found in [these lines of the source code|
https://github.com/explosion/spaCy/blob/master/spacy/pipeline/pipes.pyx#L1124-L1133]. Basically, `labels` is a `property` set by taking `label = move.split("-")[1]` for each `move in model.entity.move_names`. So you should remove hyphens or `model.entity.labels` will be broken: can you do that? 

> The third list is just wrong (it is missing PATHOLOGICAL_FORMATION and ORGANISM_SUBSTANCE), and we need to fix that, thanks!

If you can fix it it would be super useful, thank you very much!

**Comment by danielkingai2** [comment]
I will likely not be fixing the issue in `nlp.entity.labels` for this version, as it would require retraining the model, but you can get the complete list from `nlp.entity.move_names` if you desire. The generic "ENTITY" should never be predicted since its not in the training data, but whether or not this impacts the model is something I will look at for the next release. And on the last point, I'll submit a fix to the github pages documentation shortly.

**Comment by danielkingai2** [comment]
I'm going to go ahead and close this. I just merged #254 which fixes the documentation, and will fix the printing issue with `nlp.entity.labels` the next time we retrain the models. The presence of the generic `ENTITY` tag is something we will have to think about more for the next time we train the models. Thanks!

**Comment by FrancescoCasalegno** [comment]
> I will likely not be fixing the issue in `nlp.entity.labels` for this version, as it would require retraining the model, but you can get the complete list from `nlp.entity.move_names` if you desire. The generic "ENTITY" should never be predicted since its not in the training data, but whether or not this impacts the model is something I will look at for the next release. And on the last point, I'll submit a fix to the github pages documentation shortly.

For anyone landing on this issue before the bug is fixed, you can get the correct labels from _any_ model using the following snippet:
```python
def get_labels(ner):
    '''Like ner.labels, but correctly handles bug in en_ner_bionlp13cg_md.'''
    return sorted({'-'.join(move.split('-')[1:]) 
                   for move in ner.move_names
                   if move[:2] in ['B-', 'I-', 'L-', 'U-']})
```
